framework/element.py

```
@@ -0,0 +1,322 @@
+        """Finds the required element.
+        Arguments:
+            timeout (int) : time to wait for element
+            raise_exception (bool): True or False
+        Returns:
+            Element: Returns self
+        """
+        try:
+            if not self.element:
+                self.element = WebDriverWait(Element.__driver, timeout).until(
+                    expected_conditions.visibility_of_element_located(self.locator)
+                )
+            return self
+        except Exception as exception:
+            if raise_exception:
+                raise NotFoundError(f"failed to find element {self.locator} with exception {exception}")
+            Element.__logger.info(f"failed to find element {self.locator} with exception {exception}")
+
+    def find_all(self, timeout=10, raise_exception=True) -> 'Element':
+        """Finds all the required elements matching locator.
+        Arguments:
+            timeout (int) : time to wait for element
+            raise_exception (bool): True or False
+        Returns:
+            Element: Returns self
+        """
+        try:
+            if not self.elements:
+                elements =  WebDriverWait(Element.__driver, timeout).until(
+                    expected_conditions.presence_of_all_elements_located(self.locator)
+                )
+                Element.__logger.info(f"found {len(elements)} elements matching {self.locator}")
+
+                self.elements = elements if elements else None
+            return self
+            
+        except Exception as exception:
+            if raise_exception:
+                raise NotFoundError(f"failed to find elements with locator: {self.locator} with exception {exception}")
+            Element.__logger.info(f"failed to find elements with locator: {self.locator} with exception {exception}")
+    
+    def get_child_element(self, child_locator: 'Element', raise_exception=True) -> 'Element':
```

This method can also return None.

---

Done, removed option to suppress exception